apps/api/src/deployment/services/deployment-reader/deployment-reader.service.ts

```diff
@@ -0,0 +1,168 @@
+    private readonly leaseHttpService: LeaseHttpService
+  ) {}
+
+  public async findByOwnerAndDseq(
+    owner: string,
+    dseq: string,
+    options?: { certificate?: { certPem: string; keyPem: string } }
+  ): Promise<GetDeploymentResponse["data"]> {
+    const deploymentResponse = await this.deploymentHttpService.findByOwnerAndDseq(owner, dseq);
+
+    if ("code" in deploymentResponse) {
+      assert(!deploymentResponse.message?.toLowerCase().includes("deployment not found"), 404, "Deployment not found");
+
+      throw new InternalServerError(deploymentResponse.message);
+    }
+
+    const { leases } = await this.leaseHttpService.listByOwnerAndDseq(owner, dseq);
+
+    const leasesWithStatus = await Promise.all(
+      leases.map(async ({ lease }) => {
+        if (!options?.certificate) {
+          return {
+            lease,
+            status: null
+          };
+        }
+
+        try {
+          const leaseStatus = await this.providerService.getLeaseStatus(
+            lease.lease_id.provider,
+            lease.lease_id.dseq,
+            lease.lease_id.gseq,
+            lease.lease_id.oseq,
+            options.certificate
+          );
+          return {
+            lease,
+            status: leaseStatus
+          };
+        } catch {
+          return {
+            lease,
+            status: null
+          };
+        }
+      })
+    );
+
+    return {
+      deployment: deploymentResponse.deployment,
+      leases: leasesWithStatus.map(({ lease, status }) => ({
+        ...lease,
+        status
+      })),
+      escrow_account: deploymentResponse.escrow_account
+    };
+  }
+
+  public async listByOwner(
+    owner: string,
+    { skip, limit }: { skip?: number; limit?: number }
+  ): Promise<{ deployments: GetDeploymentResponse["data"][]; total: number; hasMore: boolean }> {
+    const pagination = skip !== undefined || limit !== undefined ? { offset: skip, limit } : undefined;
+    const deploymentReponse = await this.deploymentHttpService.loadDeploymentList(owner, "active", pagination);
+    const deployments = deploymentReponse.deployments;
+    const total = parseInt(deploymentReponse.pagination.total, 10);
+
+    const { results: leaseResults } = await PromisePool.withConcurrency(100)
+      .for(deployments)
+      .process(async deployment => this.leaseHttpService.listByOwnerAndDseq(owner, deployment.deployment.deployment_id.dseq));
+
+    const deploymentsWithLeases = deployments.map((deployment, index) => ({
+      deployment: deployment.deployment,
+      leases:
+        leaseResults[index]?.leases?.map(({ lease }) => ({
+          ...lease,
+          status: null as null
+        })) ?? [],
+      escrow_account: deployment.escrow_account
+    }));
+
+    return {
+      deployments: deploymentsWithLeases,
+      total,
+      hasMore: skip !== undefined && limit !== undefined ? total > skip + limit : false
+    };
+  }
+
+  public async listByOwnerAndStatus({
```

Comment:
non-blocking: the return type of this method is different then the one above. From the name I'd assume it may be the same. From the code it seems like it's summarising resources somehow. Perhaps it makes sense to reflect it in the name?

I know this is a compound from different sources, ideally we'd want to use smth more unified eventually.